core/vm/operations_acl.go

```diff
@@ -242,3 +243,108 @@ func makeSelfdestructGasFn(refundsEnabled bool) gasFunc {
+		var (
+			total uint64 // total dynamic gas used
+			addr  = common.Address(stack.Back(1).Bytes20())
+		)
+
+		// Check slot presence in the access list
+		if !evm.StateDB.AddressInAccessList(addr) {
+			evm.StateDB.AddAddressToAccessList(addr)
+			// The WarmStorageReadCostEIP2929 (100) is already deducted in the form of a constant cost, so
+			// the cost to charge for cold access, if any, is Cold - Warm
+			coldCost := params.ColdAccountAccessCostEIP2929 - params.WarmStorageReadCostEIP2929
+			// Charge the remaining difference here already, to correctly calculate available
+			// gas for call
+			if !contract.UseGas(coldCost, evm.Config.Tracer, tracing.GasChangeCallStorageColdAccess) {
+				return 0, ErrOutOfGas
+			}
+			total += coldCost
+		}
+
+		// Check if code is a delegation and if so, charge for resolution.
+		if target, ok := types.ParseDelegation(evm.StateDB.GetCode(addr)); ok {
+			var cost uint64
+			if evm.StateDB.AddressInAccessList(target) {
+				cost = params.WarmStorageReadCostEIP2929
+			} else {
+				evm.StateDB.AddAddressToAccessList(target)
+				cost = params.ColdAccountAccessCostEIP2929
+			}
+			if !contract.UseGas(cost, evm.Config.Tracer, tracing.GasChangeCallStorageColdAccess) {
+				return 0, ErrOutOfGas
+			}
+			total += cost
+		}
+
+		// Now call the old calculator, which takes into account
+		// - create new account
+		// - transfer value
+		// - memory expansion
+		// - 63/64ths rule
+		old, err := oldCalculator(evm, contract, stack, mem, memorySize)
+		if err != nil {
+			return old, err
+		}
+
+		// Temporarily add the gas charge back to the contract and return value. By
+		// adding it to the return, it will be charged outside of this function, as
+		// part of the dynamic gas. This will ensure it is correctly reported to
+		// tracers.
+		contract.Gas += total
+
+		var overflow bool
+		if total, overflow = math.SafeAdd(old, total); overflow {
+			return 0, ErrGasUintOverflow
+		}
+		return total, nil
+	}
+}
+
+func gasEip7702CodeCheck(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
+	cost, _ := gasEip2929AccountCheck(evm, contract, stack, mem, memorySize)
+	// Check if code is a delegation and if so, charge for resolution
+	addr := common.Address(stack.peek().Bytes20())
+	if addr, ok := types.ParseDelegation(evm.StateDB.GetCode(addr)); ok {
+		if evm.StateDB.AddressInAccessList(addr) {
+			cost += params.WarmStorageReadCostEIP2929
+		} else {
+			evm.StateDB.AddAddressToAccessList(addr)
+			cost += params.ColdAccountAccessCostEIP2929
+		}
+	}
+	return cost, nil
+}
+
+func gasExtCodeCopyEIP7702(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
+	gas, err := gasExtCodeCopyEIP2929(evm, contract, stack, mem, memorySize)
+	if err != nil {
+		return 0, err
+	}
+	// Check if code is a delegation and if so, charge for resolution
+	addr := common.Address(stack.peek().Bytes20())
```

Comment:
According to EIP, 

```
If a code reading instruction accesses a cold account during the resolution of delegated code, add an additional [EIP-2929](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2929.md) COLD_ACCOUNT_READ_COST cost of 2600 gas to the normal cost and add the account to accessed_addresses. Otherwise, assess a WARM_STORAGE_READ_COST cost of 100.
```

The question here is: the code of delegation designator is not resolved by `ExtCodeCopy`, EIP `and requires all other code reading operations to act only on the first 2 bytes of the designator (0xef01).`

Maybe it should be clarified a bit, if it's interpreted as resolution or not.

Reply:
It should be cleared up in the EIP here: https://github.com/ethereum/EIPs/pull/9136#pullrequestreview-2505128829

Reply:
I removed the dynamic gas accounting for the code *reading* ops here [e4e381d](https://github.com/ethereum/go-ethereum/pull/30078/commits/e4e381d70ac11a8cf10722770d67d40bfb8fa6aa).

There should be no change in cost from the previous version since the delegation is no longer resolved.